README.md

```diff
@@ -28,41 +27,129 @@ This can be useful in several ways:
-`./idpbuilder create --kube-version v1.27.3`
+* ArgoCD is the GitOps solution to deploy manifests to Kubernetes clusters. In this project, a package is an ArgoCD application. 
+* Gitea server is the in-cluster Git server that ArgoCD can be configured to sync resources from. You can sync from local file systems to this.
+* Ingress-nginx is used as a method to access in-cluster resources such as ArgoCD UI and Gitea UI.
 
-If it is needed to expose some extra Ports between the docker container and the kubernetes host, they can be declared as such
-`./idpbuilder create --extra-ports 22:32222`
+See the [Architecture](#Architecture) section for more information on further information on how core packages are installed and configured.
 
-It is also possible to use your own kind config file
-`./idpbuilder create --build-name local --kind-config ./my-kind.yaml`
+Once idpbuilder finishes provisioning cluster and packages, you can access GUIs by going to the following addresses in your browser.
 
-**NOTE**: Be sure to include in your kind config the section `containerdConfigPatches` where the registry hostname includes the name specified with the parameter: `--build-name`
-```yaml
-containerdConfigPatches:
-- |-
-  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:5001"]
-    endpoint = ["http://idpbuilder-<localBuildName>-registry:5000"]
+* ArgoCD: https://argocd.cnoe.localtest.me:8443/
+* Backstage: https://backstage.cnoe.localtest.me:8443/
+* Gitea: https://gitea.cnoe.localtest.me:8443/
+
+You can obtain credentials for these packages by running the following commands:
+
+```bash
+# argoCD password. username is admin.
+kubectl -n argocd get secret argocd-initial-admin-secret \
+  -o go-template='{{ range $key, $value := .data }}{{ printf "%s: %s\n" $key ($value | base64decode) }}{{ end }}'
+
+# gitea admin credentials
+kubectl get secrets -n gitea gitea-admin-secret \
+  -o go-template='{{ range $key, $value := .data }}{{ printf "%s: %s\n" $key ($value | base64decode) }}{{ end }}'
 ```
 
 ### Use
 
-#### GUI
+Run the following command to see available flags and sub commands
 
-GUIs for core packages are available at the following addresses:  
+```bash
+# for general command information
+./idpbuilder --help 
+
+# for sub command specific information
+./idpbuilder create --help
+```
+
+#### Examples
+
+You can specify the kubernetes version by using the `--kube-version` flag. Supported versions are available [here](https://github.com/kubernetes-sigs/kind/releases).
+
+```
+./idpbuilder create --kube-version v1.27.3
+```
+
+If you need to expose more ports between the docker container and the kubernetes host, you can use the `--extra-ports` flag. For example:
+
+```
+./idpbuilder create --extra-ports 22:32222`
+```
+
+If you want to specify your own kind configuration file, use the `--kind-config` flag.
+
+```
+./idpbuilder create --build-name local --kind-config ./my-kind.yaml`
```

Comment:
We don't mention `--build-name` by this point. Might want to remove this or add the help text about it right above this.